p2p/libp2p/networksharding/kadSharder.go

```diff
@@ -69,11 +69,31 @@ func (ks *kadSharder) GetDistance(a, b sortingID) *big.Int {
 }
 
 // SortList sort the provided peers list
-func (ks *kadSharder) SortList(peers []peer.ID, ref peer.ID) []peer.ID {
-	return sortList(ks, peers, ref)
+func (ks *kadSharder) SortList(peers []peer.ID, ref peer.ID) ([]peer.ID, bool) {
+	sl := getSortingList(ks, peers, ref)
+	// for balance we should have between 1 and 20% connections outside of shard
+	peerCnt := len(peers)
+	inShardCnt := inShardCount(sl)
+	balanced := peerCnt > inShardCnt
+
+	if balanced && peerCnt > 5 {
```

Comment:
I do not understand your magic numbers
why multiply by 4 and divide by 5... to get 20 %?
the percentages should be configurable, worst case have them in constants on top of the file.